tools/metilene/metilene.xml

```diff
@@ -0,0 +1,158 @@
+    ##echo ",".join(map(str,$opt_in2))
+    >/dev/null 2>/dev/null
+    ;
+    ## ===============  metilene main program
+    metilene 
+    -M $Options.opt_M -m $Options.opt_m -d $Options.opt_d -t 1 -f 1 -v $Options.opt_v
+    metilene_g1_g2.input
+    >$output_bedlike
+    2>/dev/null
+    ;
+    ## =============== postprocessing
+    cut -f1-4 $output_bedlike > $output_bedgraph 2>/dev/null
+    ;
+    ## write to job directory ...
+    metilene_output.pl -q $output_bedlike -p 0.05 -o ./metilene >/dev/null 2>/dev/null
+    ;
+    ## ... and move to output files
+    mv metilene_qval.0.05.bedgraph $output_qval_bedgraph 2>/dev/null;
+    mv metilene_qval.0.05.out $output_qval_out 2>/dev/null;
+    mv metilene_qval.0.05.pdf $output_qval_pdf 2>/dev/null
+    ]]></command>
+    
+    <inputs>
+        <!-- input bedgraph files 1 -->
+        <param
+            name="opt_in1"
+            type="data" format="bed,bedgraph" multiple="true"
+            optional="false"
+            label="Input group 1"
+            help="set of sorted bedgraph input profiles of group 1" />
+        <!-- input bedgraph files 2 -->
+        <param
+            name="opt_in2"
+            type="data" format="bed,bedgraph" multiple="true"
+            optional="false"
+            label="Input group 2"
+            help="set of sorted bedgraph input profiles of group 2" />
+        
+        <!-- Advanced parameters (folded) -->
+        <section name="Options" title="Advanced Options" expanded="False">
+	    <param
+                name="opt_M" 
+                label="The allowed nt distance between two CpGs within a DMR"
+                type="integer"
+	        optional="false"
+                value="300"
+	        help="-M, --maxdist"/>
+
+            <param
+                name="opt_m" 
+                label="The minimum # of CpGs in a DMR"
+                type="integer"
+	        optional="false"
+                value="10"
+	        help="-m, --mincpgs"/>
+            
+            <param
+                name="opt_d" 
+                label="The minimum mean methylation difference for calling DMRs"
+                type="float"
+	        optional="false"
+                value="0.1"
+	        help="-d, --minMethDiff"/>
+
+            <param
+                name="opt_v" 
+                label="Stringency of the valley filter (0.0 - 1.0)"
+                type="float"
+	        optional="false"
+                value="0.7"
+	        help="-v, --valley">
+                <validator type="in_range" min="0.0" max="1.0" 
+                           message="Valley filter value must be in [0,1]." />
+            </param>    
+	</section>
+    </inputs>
+    
+    <outputs>
+        <!-- finally: for bed-like use format=tabular -->
+        <data format="tabular" name="output_bedlike" 
+              label="${tool.name} on ${on_string}" />
+        <data format="bedgraph" name="output_bedgraph" 
+              label="${tool.name} on ${on_string}" />
```

Comment:
I would keep this for uniformity, (if you don't mind.) The second output (output_bedgraph) could use an additional qualification anyway.